### PR TITLE
submission of completed assessment

### PR DESCRIPTION
**Pull request by csaument**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -32,11 +43,49 @@ const object2 = { number: 0 }
...
+      }
+      if (object.number % 3 === 0) {
+            return `${object.number} is divisible by three`
+      } else if (Math.abs(object.number % 3) > 0) {
+            return `${object.number} is not divisible by three`
+      } else {
+            throw Error("Something went wrong")
+      }
+}
+
```
Comment (Genemichael25) [inline code-challenges.test.js:72]: Great job thinking of those errors and ensuring only an object is passed!

File: code-challenges.test.js [hunk 2]
```diff
@@ -57,3 +140,38 @@ const vowelTester3 = "challenges"
...
+            newWord.indexOf("o"),
+            newWord.indexOf("u"),
+            newWord.indexOf("y")].filter(value => value >= 0)
+      if (indexes.length === 0) {
+            return -1
+      } else {
+            return Math.min(...indexes)
+      }
+}
+
```
Comment (Genemichael25) [inline code-challenges.test.js:173]: Very cool and different approach! You do an excellent job of thinking about edge cases and making sure you control the inputs for your functions!

File: coding-concepts.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ const myCodingSkills = {
   databases: "PostgreSQL",
   versionControl: "GitHub"
 }
-// console.log(myCodingSkills.languages[0])
+console.log(myCodingSkills.languages[0])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: "JavaScript"
+// b) Verify and explain: myCodingSkills is an object. The languages key is mapped to an array, ["JavaScript", "Ruby"], which is accessed directly through the syntax object.key. The [0] accesses the first element in this array, which is passed ot the console log.
```
Comment (Genemichael25) [inline coding-concepts.js:53]: Great job on this portion, really specific explanation of what we see in the console logs!👍🏽

File: interview-questions.md [hunk 4]
```diff
@@ -9,49 +9,133 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+The .map() method is used to manipulate each element within an array. This method always returns an array of the same length. The basic syntax array.map(value => *manipulation*) allows each value to undergo the designated manipulation.
+
+The .filter() method is used to return a subset of the original array, based upon criteria designated by the nested function. The basic syntax array.filter(value => *criteria*) returns an array with the values where criteria are fulfilled.
+
+Notably, the .filter() method does not modify the values extracted from the original array. Nesting a .filter() method within a .map() method through .filter().map() would allow manipulation of the values contained in the subset of the original array.
+
+Although reversing the order of nesting these methods is possible in concept (ie .map().filter()), such manipulation is redundant with the functionality inherent to the .filter() method.
+
```
Comment (Genemichael25) [inline interview-questions.md:45]: Wonderful insight into .map and .filter !

File: coding-concepts.js [hunk 5]
```diff
@@ -8,36 +8,36 @@
 // --------------------1) What will this log?
 
 const cohort = "Alpha 2023"
-// console.log(cohort.split(""))
+console.log(cohort.split(""))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: ["A", "l", "p", "h", "a", " ", "2", "0", "2", "3"]
+// b) Verify and explain: .split() is a built-in function for strings that divides the string into substring elements within an array. The "" argument, when passed to the function, serves to divide the string at each character.
```
Comment (Genemichael25) [inline coding-concepts.js:14]: Awesome explanation of the "" and how that works!

**Review by Genemichael25 — APPROVED** [review]
Outstanding work Chris! You continue to demonstrate a competent level of understanding. I love the fact that you are thoughtful in your coding and account for cases we don't ask you for. Keep up this level of performance!
